app/src/components/form/attribute_input.rs

```diff
@@ -68,3 +69,59 @@ pub fn single_attribute_input(props: &SingleAttributeInputProps) -> Html {
         </div>
     }
 }
+
+#[derive(Properties, PartialEq)]
+pub struct ListAttributeInputProps {
+    pub name: String,
+    pub attribute_type: AttributeType,
+    pub values: Vec<String>,
+}
+
+#[function_component(ListAttributeInput)]
+pub fn list_attribute_input(props: &ListAttributeInputProps) -> Html {
+
+    // let value_indices = use_memo(props.values, |vals| use_state(|| (0..(vals.len() as isize)).collect::<Vec<_>>()));
+    let value_indices = use_state(|| (0..(props.values.len() as isize)).collect::<Vec<_>>() ); // use_memo(props.values)
+    let new_index = use_mut_ref::<isize>(|| 0);
+
+    html! {
+        <div class="row mb-3">
+            <label for={props.name.clone()}
+                class="form-label col-4 col-form-label"
+                title={props.name.clone()}>
+                {props.name[0..1].to_uppercase() + &props.name[1..].replace("_", " ")}{":"}
```

Comment:
Extract the name formatting to a function?

Reply:
Oh, duh! I didn't even realize I had duplicated that code. :sweat_smile:

Reply:
It could still use a function, but at least it's not duplicated now (: